app/src/pages/Lobby/index.tsx

```diff
@@ -41,181 +32,29 @@ const useStyles = makeStyles((theme: Theme) =>
-            </Grid>
-            <Grid item>
-              <ScreenCardsCheckbox value={Boolean(currentGame.screen_cards)} />
-            </Grid>
-
-            <Grid item>
-              <Typography variant="h6" className={titleClasses.title}>
-                Turns
-              </Typography>
-            </Grid>
-            <Grid item>
-              <SecondsPerTurnInput
-                value={String(currentGame.seconds_per_turn || "")}
-              />
-            </Grid>
-            <Grid item>
-              <AllowCardSkipsCheckbox
-                value={Boolean(currentGame.allow_card_skips)}
-              />
-            </Grid>
-            <Grid item>
-              <Typography variant="h6" className={titleClasses.title}>
-                Rounds
-              </Typography>
-              <Box pl={2} pt={1} fontSize="0.75rem" color={grey[600]}>
-                {currentPlayer.role === PlayerRole.Host &&
-                  "You can add, remove, or reorder rounds. By default, cards submitted will be re-used across rounds of Taboo, Charades, and Password."}
-              </Box>
-            </Grid>
-            <Grid item>
-              {currentPlayer.role === PlayerRole.Host ? (
-                <ControllableRoundSettings />
-              ) : (
-                <RoundSettings />
-              )}
-            </Grid>
-            <Grid item />
-          </Grid>
-        </Grid>
-      </ExpansionPanelDetails>
-    </ExpansionPanel>
-  )
-}
-
-function WaitingRoomSection() {
-  const currentPlayer = React.useContext(CurrentPlayerContext)
-  const titleClasses = useTitleStyle()
-
-  return (
-    <Grid item>
-      <Grid container spacing={2} direction="column">
-        <Grid item>
-          <Typography variant="h4" className={titleClasses.title}>
-            Lobby
-          </Typography>
-        </Grid>
-        <Grid item>
-          <UsernameInput
-            username={currentPlayer.username || ""}
-            playerId={currentPlayer.id}
-          />
-        </Grid>
-        <WaitingRoom></WaitingRoom>
-      </Grid>
-    </Grid>
-  )
-}
-
 function Lobby() {
   const currentPlayer = React.useContext(CurrentPlayerContext)
+  const currentGame = React.useContext(CurrentGameContext)
   const [updatePlayer] = useUpdatePlayerMutation()
   const classes = useStyles()
   const titleClasses = useTitleStyle()
   const location = useLocation()
   const [hideShare, setHideShare] = React.useState(false)
+  const [updateGameSettings] = useUpdateGameSettingsMutation()
+  const [cardPlayStyle, setCardPlayStyle] = React.useState(
+    GameCardPlayStyleEnum.PlayersSubmitWords
+  )
+
+  React.useEffect(() => {
+    setCardPlayStyle(currentGame.card_play_style)
+  }, [currentGame.card_play_style])
+
+  const [wordList, setWordList] = React.useState("")
+  const debouncedSetWordList = React.useRef(
+    debounce((value: string) => {
+      setWordList(value)
+    }, DEBOUNCE_SECONDS)
+  ).current
```

Comment:
since this is being set a few components down (but state is kept at this level for WaitingRoom, which is a sibling to CardSEttings), typing a list of words in the input form was sluggish. 

While I imagine most folks will copy paste a list in (therefore just causing one render) - this is a solution to make the experience smooth for those typing or editing the comma separated list

Reply:
> While I imagine most folks will copy paste a list in

I wonder if newline delimitation would be helpful for this use case.

Reply:
I was thinking about it for sure - I'll see how it does in the field. Likely could support both and intelligently detect the delimeter.